chain-tx-enclave-next/mls/src/tree.rs

```diff
@@ -350,7 +328,37 @@ impl Tree {
         Ok(Self {
             nodes: vec![Node::Leaf(Some(creator_kp))],
             cs,
-            my_pos: 0,
+            my_pos: LeafSize(0),
         })
     }
 }
+
+fn node_hash(nodes: &[Node], cs: CipherSuite, index: NodeSize, leaf_size: LeafSize) -> Vec<u8> {
+    let node = &nodes[index.0];
+    match node {
+        Node::Leaf(kp) => {
+            let mut inp = Vec::new();
+            LeafNodeHashInput {
+                node_index: index.0 as u32,
+                key_package: kp.clone(),
+            }
+            .encode(&mut inp);
+            cs.hash(&inp)
+        }
+        Node::Parent(pn) => {
+            let left_index = index.left().unwrap();
+            let left_hash = node_hash(nodes, cs, left_index, leaf_size);
+            let right_index = index.right(leaf_size).unwrap();
+            let right_hash = node_hash(nodes, cs, right_index, leaf_size);
+            let mut inp = Vec::new();
+            ParentNodeHashInput {
+                node_index: index.0 as u32,
+                parent_node: pn.clone(),
+                left_hash,
+                right_hash,
+            }
+            .encode(&mut inp);
```

Comment:
nit: `.get_encoding()` is a short-hand for this, so could be:
```
let inp = ParentNodeHashInput {
                  node_index: index.0 as u32,
                  parent_node: pn.clone(),
                  left_hash,
                  right_hash,
              }.get_encoding();
```
?

Reply:
done